packages/syft/src/syft/service/worker/utils.py

```diff
@@ -254,9 +257,46 @@ def run_workers_in_threads(
     return results
 
 
+def prepare_worker_pool_env(runner: KubernetesRunner, env_vars: dict):
+    # get current backend pod name
+    backend_pod_name = os.getenv("K8S_POD_NAME")
+    if not backend_pod_name:
+        raise ValueError(message="Pod name not provided in environment variable")
+
+    # get current backend's credentials path
+    creds_path = os.getenv("CREDENTIALS_PATH")
+    if not creds_path:
+        raise ValueError(message="Credentials path not provided")
+
+    creds_path = Path(creds_path)
+    if not creds_path.exists():
+        raise ValueError(message="Credentials file does not exist")
+
+    # create a secret for the node credentials owned by the backend, not the pool.
+    node_secret = KubeUtils.create_secret(
```

Comment:
We could add a TODO or a github issue, where we are able to remove the credentials.json creation at backend using Init containers which we discussed or any other methods, for later cycles.

Reply:
https://github.com/OpenMined/Heartbeat/issues/975